src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -60,26 +62,35 @@ public class PullRequestHandler : IHttpHandler
             context.Response.Write(HttpUtility.HtmlEncode(result));
             return;
         }
-        else
+
+        success = false;
+
+        // Check inputs to the script. PR has precedence
+        uint pr;
+        if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
         {
-            // Check inputs to the script
-            uint pr;
-            if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
-            {
-                success = SyncToPr(pr, result);
-            }
-            else
+            success = SyncToPr(pr, result);
+        }
+        else if (!string.IsNullOrEmpty(branchString))
+        {
+            // We only want to support limited branches as we only use this for PR purposes
+            switch (branchString)
             {
-                context.Response.Write("Invalid PR ID specified<br/>");
-                success = false;
+                case "master":
+                    success = SyncToBranch(branchString, result);
+                    break;
```

Comment:
If you support other branches in the future, do you expect people calling a function other than `SyncToBranch`? If the answer is "no", then here should not be a `switch` but really a `if (branchstring=="master")` clause for today and it will go away when you support branches in general.